core/arch/arm/dts/stm32mp131.dtsi

```diff
@@ -83,6 +83,16 @@
 
 	};
 
+	sdmmc1_io: sdmmc1_io {
+		compatible = "st,stm32mp13-iod";
+		regulator-name = "sdmmc1_io";
```

Comment:
+		regulator-always-on;

you need  to define the IO domain as always on regulator in STM32MP13 SoC dtsi, to enable
IOCOMP  and  deactivate the level detectors in SYSCFG

in most of case the input regulator can't be stopeed (only allow 3V3 to 1V8 switch).

For "always-on" regulator, it is enabled during the IOD regulator probe 
It will be never deactivated by regulator framework (it is no required by Linux = VQMMC)  
and it is no issue to keep this feature enabled.

The ICOMP init is no more done in SYSCFG for IOD managed in driver stm32mp1_regulator_iod.c 
see test  on SYSCFG_IO_COMP_IDX_SD1 / SYSCFG_IO_COMP_IDX_SD2 in syscfg driver

it is done in SYSCFG driver  ONLY for IOCOMP linked to VDD.

Reply:
Right, thanks. I confused the SDMMC device regulator supply and the SDMMC IO domain regulator supply.